Feature/core 500 add profile picture component

karladler opened the pull request:
add general profile component.
Upload interface is open for discussion

karladler committed:
CORE-500 introduce profile picture component

karladler committed:
CORE-500 resolve conflict, add copyright

Commit:
Merge branch 'develop' into feature/CORE-500-add-profile-picture-component

Commit:
CORE-500

Draft Profile-Picture

karladler committed:
CORE-500 add image masks

css rage magic

karladler committed:
CORE-500 allow other size units than px

Commit:
change comment

- [CORE-500]

karladler committed:
change size keywords to comply with bootstrap

[CORE-500]

Commit:
add verification icon

- [CORE-500]

Commit:
Merge branch 'feature/CORE-500-add-profile-picture-component' of github.com:evannetwork/ui-vue into feature/CORE-500-add-profile-picture-component

# Conflicts:
#	dapps/evancore.vue.libs/src/components/profile-picture/profile-picture.scss

Commit:
rename to verification.svg

- [CORE-500]

Commit:
scale verification icon depends on size prop

-[CORE-500]

karladler committed:
add first draft of file upload component

[CORE-500]

Commit:
add logic for verification icon

- display verification icon depends on isVerified prop
- change position of icon for unknown and user type
-[CORE-500]

Commit:
Merge branch 'feature/CORE-500-add-profile-picture-component' of github.com:evannetwork/ui-vue into feature/CORE-500-add-profile-picture-component

# Conflicts:
#	dapps/evancore.vue.libs/src/components/profile-picture/profile-picture.vue

Commit:
add display logic for upload button

-[CORE-500]

karladler committed:
cleanup code, use clipping path for masking

[CORE-500]

Tschuck commented on dapps/evancore.vue.libs/src/components/profile-picture/profile-picture.scss:
```diff
@@ -0,0 +1,123 @@
+/*
```
Should we move this to ui-core general css style? So it could be used by other frameworks.

Tschuck commented on dapps/evancore.vue.libs/src/components/profile-picture/profile-picture.ts:
```diff
@@ -0,0 +1,99 @@
+  /**
+   * ui.libs evan dapp base url
+   */
+  uiBaseUrl = '';
```
[uiLibBaseUrl](https://github.com/evannetwork/ui-vue/blob/develop/dapps/evancore.vue.libs/src/vue-core.ts#L75) is generally loaded within the vue-core, you can use this one ;)

`$store.state.uiLibBaseUrl`

Tschuck commented on dapps/evancore.vue.libs/src/components/profile-picture/profile-picture.vue:
```diff
@@ -0,0 +1,72 @@
+    <!-- File upload modal -->
+    <evan-modal ref="pictureUploadModal" :maxWidth="'600px'">
+      <template v-slot:header>
+        <h5 class="modal-title">{{'Upload your profile picture here' || translate}}</h5>
```
translation missing

Tschuck commented on dapps/evancore.vue.libs/src/components/profile-picture/profile-picture.ts:
```diff
@@ -0,0 +1,99 @@
+      error: false,
+      setDirty: () => {}
+    }
+  }) fileForm: EvanFormControl; // TODO: check whether this is the correct interface
```
The `fileForm` should be replaced with a `picture` property, that can be a `string` or a loaded `file`. On change we can send an `pictureChange` event, that can be handled by the using component.

The `img` `src` should handle the defined `picture` property, so the property interface will be correct.

I added a very quick and dirty test implementation as [gist](https://gist.github.com/Tschuck/d1582b556d3947e022bc1ea5bd5ae01f).

Tschuck reviewed (changes requested):
Please have a look into the comments ;)

When the changes are done, you can already add the component to the `profile-preview` component and replace the placeholder.

karladler committed:
implement PR fixes

[CORE-500]
- use baseUrl from store,
- add translation

karladler committed:
add image placeholder and initials

- add image placeholder, when no src is given
- use first two characters as initials, when no image is uploaded
- [CORE-500]

karladler committed:
add translations, fix upload function

karladler committed:
fix upload handling logic

- event should be only thrown when change function is called
- [CORE-500]

karladler committed:
rename alias and name into accountName

include prfile picture component in preview

karladler committed:
Merge branch 'develop' into feature/CORE-500-add-profile-picture-component

# Conflicts:
#	dapps/evancore.vue.libs/src/components/profile-preview/profile-preview.ts
#	dapps/evancore.vue.libs/src/components/profile-preview/profile-preview.vue
